modules/helfi_toc/assets/js/tableOfContents.js

```diff
@@ -32,62 +34,149 @@
+        '5': '[⁵₅۵٥５]',
+        '6': '[⁶₆۶٦６]',
+        '7': '[⁷₇۷７]',
+        '8': '[⁸₈۸８]',
+        '9': '[⁹₉۹９]',
+        'a': '[àáảãạăắằẳẵặâấầẩẫậāąåαάἀἁἂἃἄἅἆἇᾀᾁᾂᾃᾄᾅᾆᾇὰᾰᾱᾲᾳᾴᾶᾷаأအာါǻǎªაअاａä]',
+        'b': '[бβبဗბｂब]',
+        'c': '[çćčĉċｃ©]',
+        'd': '[ďðđƌȡɖɗᵭᶁᶑдδدضဍဒდｄᴅᴆ]',
+        'e': '[éèẻẽẹêếềểễệëēęěĕėεέἐἑἒἓἔἕὲеёэєəဧေဲეएإئｅ]',
+        'f': '[фφفƒფｆ]',
+        'g': '[ĝğġģгґγဂგگｇ]',
+        'h': '[ĥħηήحهဟှჰｈ]',
+        'i': '[íìỉĩịîïīĭįıιίϊΐἰἱἲἳἴἵἶἷὶῐῑῒῖῗіїиဣိီည်ǐიइیｉi̇ϒ]',
+        'j': '[ĵјჯجｊ]',
+        'k': '[ķĸкκقكကკქکｋ]',
+        'l': '[łľĺļŀлλلလლｌल]',
+        'm': '[мμمမმｍ]',
+        'n': '[ñńňņŉŋνнنနნｎ]',
+        'o': '[óòỏõọôốồổỗộơớờởỡợøōőŏοὀὁὂὃὄὅὸόоوθိုǒǿºოओｏöө]',
+        'p': '[пπပპپｐ]',
+        'q': '[ყｑ]',
+        'r': '[ŕřŗрρرრｒ]',
+        's': '[śšşсσșςسصစſსｓŝ]',
+        't': '[ťţтτțتطဋတŧთტｔ]',
+        'u': '[úùủũụưứừửữựûūůűŭųµуဉုူǔǖǘǚǜუउｕўü]',
+        'v': '[вვϐｖ]',
+        'w': '[ŵωώဝွｗ]',
+        'x': '[χξｘ]',
+        'y': '[ýỳỷỹỵÿŷйыυϋύΰيယｙῠῡὺ]',
+        'z': '[źžżзζزဇზｚ]',
+        'aa': '[عआآ]',
+        'ae': '[æǽ]',
+        'ai': '[ऐ]',
+        'ch': '[чჩჭچ]',
+        'dj': '[ђđ]',
+        'dz': '[џძ]',
+        'ei': '[ऍ]',
+        'gh': '[غღ]',
+        'ii': '[ई]',
+        'ij': '[ĳ]',
+        'kh': '[хخხ]',
+        'lj': '[љ]',
+        'nj': '[њ]',
+        'oe': '[öœؤ]',
+        'oi': '[ऑ]',
+        'oii': '[ऒ]',
+        'ps': '[ψ]',
+        'sh': '[шშش]',
+        'shch': '[щ]',
+        'ss': '[ß]',
+        'sx': '[ŝ]',
+        'th': '[þϑثذظ]',
+        'ts': '[цცწ]',
+        'ue': '[ü]',
+        'uu': '[ऊ]',
+        'ya': '[я]',
+        'yu': '[ю]',
+        'zh': '[жჟژ]',
+        'gx': '[ĝ]',
+        'hx': '[ĥ]',
+        'jx': '[ĵ]',
+      };
+
       // Craft table of contents.
       once('table-of-contents', titleComponents.join(','), mainContent)
         .forEach(function (content) {
-          const name = content.textContent
+          let name = content.textContent
             .toLowerCase()
-            .trim()
-            .replace(/ä/gi, 'a')
-            .replace(/ö/gi, 'o')
-            .replace(/å/gi, 'a')
-            .replace(/\W/g, '-')
-            .replace(/-(\d+)$/g, '_$1');
-
-          let nodeName = content.nodeName.toLowerCase();
-          if (nodeName === 'button') {
-            nodeName = content.parentElement.nodeName.toLowerCase();
+            .trim();
+
+        // To ensure backwards compatibility, this is done only to "other" languages.
+        if (!mainLanguages.includes(drupalSettings.path.currentLanguage)) {
+          Object.keys(swaps).forEach((swap) => {
+            name = name.replace(new RegExp(swaps[swap], 'g'), swap);
+          });
           }
+          else {
+            name = name
+              .replace(/ä/gi, 'a')
+              .replace(/ö/gi, 'o')
+              .replace(/å/gi, 'a');
+        }
+
+          name = name.replace(/\W/g, '-').replace(/\s/g, '-').replace(/-(\d+)$/g, '_$1');
```

Comment:
This line should have a brief explanation. It's not obvious (at least to me) what is being done here without running the actual regex.

Reply:
Added comment. I also removed `.replace(/\s/g, '-')` since it is redundant. `\W` is equivalent to `[^a-zA-Z0-9_]`.